src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,54 @@
+package duke; 
+
+import duke.task.TaskList; 
+import duke.storage.Storage; 
+import duke.ui.Ui;
+
+public class Duke {
+    private TaskList tasks;
+    private Ui ui;
+    private Storage storage;
+
+    public Duke() {
+        this.storage = new Storage();
+        this.tasks = new TaskList(storage.loadTasks());
+        this.ui = new Ui();
+    }
+
+    public void run() {
+        ui.printHello();
+        String command = ui.getCommand();
+
+        while (!command.equals("bye")) {
+            try {
+                String[] split = command.split(" ", 2);
+
+                switch (split[0]) {
+                case "list":
+                    ui.printTasks(tasks);
+                    break;
+                case "done":
+                    tasks.markTaskAsDone(Integer.parseInt(split[1]) - 1);
```

Comment:
Something to consider: a reader won't know the reasoning/context behind this number